tests/model_monitoring/db/tsdb/timescaledb/test_timescaledb_connector.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for TimescaleDBConnector database auto-creation functionality."""
+
+import contextlib
+import uuid
+
+import pytest
+
+
+class TestTimescaleDBConnectorDatabaseCreation:
+    """Test database auto-creation functionality in TimescaleDBConnector."""
+
+    @pytest.fixture
+    def test_system_id(self):
+        """Generate unique system_id for test isolation."""
+        return f"test_{uuid.uuid4().hex[:8]}"
+
+    @pytest.fixture
+    def profile(self, connection_string):
+        """Create profile from connection string."""
+        from mlrun.datastore.datastore_profile import DatastoreProfilePostgreSQL
+
+        return DatastoreProfilePostgreSQL.from_dsn(
+            dsn=connection_string, profile_name="test_profile"
+        )
+
+    def _force_drop_database(self, admin_connection, database_name):
+        """Force drop a database by terminating all connections first.
+
+        Silently ignores all errors to ensure cleanup doesn't fail tests.
+        """
+        from mlrun.model_monitoring.db.tsdb.timescaledb.timescaledb_connection import (
+            Statement,
+        )
+
+        # Terminate all connections to the database
+        with contextlib.suppress(Exception):
+            admin_connection.run(
+                query=Statement(
+                    sql="""
+                    SELECT pg_terminate_backend(pid)
+                    FROM pg_stat_activity
+                    WHERE datname = %s AND pid <> pg_backend_pid()
+                    """,
+                    parameters=(database_name,),
+                )
+            )
+        # Now drop the database
+        with contextlib.suppress(Exception):
+            admin_connection.run(
+                statements=[f'DROP DATABASE IF EXISTS "{database_name}"']
+            )
```

Comment:
Might want to consider logging the error, or we'll never know about it.

Reply:
Intentional - the docstring notes "Silently ignores all errors to ensure cleanup doesn't fail tests." For test cleanup, I prefer silent suppression to avoid noisy output. If cleanup consistently fails, we'd notice from leftover databases or subsequent test failures.

Reply:
Fair point. Changed to log warnings on cleanup failures - this way tests stay clean but debugging info is available when things go wrong.